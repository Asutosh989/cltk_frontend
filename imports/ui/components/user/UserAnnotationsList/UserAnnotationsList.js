import Annotations from '/imports/collections/annotations';

UserAnnotationsList = React.createClass({

	mixins: [ReactMeteorData],

	getMeteorData() {
		let annotatedText = [];
		const handleAnnotation = Meteor.subscribe('annotation');
		if (handleAnnotation.ready()) {
			annotatedText = Annotations.find({ user: Meteor.userId() }).fetch();
		}
		return {
			annotatedText,
		};
	},

	render() {
		styles = {
			listItem: {
				overflow: 'hidden',
				textOverflow: 'ellipsis',
				whiteSpace: 'nowrap',
				font: 'normal',
			},
			innerList: {
				maxHeight: 250,
				overflow: 'auto',
			},
			list: {
				margin: 0,
			},
		};

		return (
			<ul className="collection with-header" style={styles.list}>
				<li className="collection-header"> <h3>Annotations</h3></li>
				<div style={styles.innerList}>
					{this.data.annotatedText.map((annotation, i) => {
						let textId = '';
						return (
							<AnnotationTextNode
								key={i}
								isOdd={i % 2 ? true : false}
								annotation={annotation}
							/>
					)})}
				</div>
			</ul>
		);
	},

});
