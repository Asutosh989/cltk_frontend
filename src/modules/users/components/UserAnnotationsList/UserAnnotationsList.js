import React from 'react';


const styles = {
	listItem: {
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		whiteSpace: 'nowrap',
		font: 'normal',
	},
	innerList: {
		maxHeight: 250,
		overflow: 'auto',
	},
	list: {
		margin: 0,
	},
};

const UserAnnotationsList = props => {
	return (
		<ul className="collection with-header" style={styles.list}>
			<li className="collection-header"> <h3>Annotations</h3></li>
			<div style={styles.innerList}>
				{props.annotatedText && props.annotatedText.map((annotation, i) => (
					<div>
						{/*
							TODO: Annotation text node
						*/}
					</div>
				))}
			</div>
		</ul>
	);
};

export default UserAnnotationsList;
