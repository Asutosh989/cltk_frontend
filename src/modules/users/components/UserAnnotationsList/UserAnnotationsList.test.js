
import React from 'react';
import { shallow } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import UserAnnotationsList from './UserAnnotationsList';

describe('UserAnnotationsList', () => {
	it('renders correctly', () => {

		const wrapper = shallow(
			<MuiThemeProvider>
				<UserAnnotationsList />
			</MuiThemeProvider>
			);
		expect(wrapper).toBeDefined();
	});
});
