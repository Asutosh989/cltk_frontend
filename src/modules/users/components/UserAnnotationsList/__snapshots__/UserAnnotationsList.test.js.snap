// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAnnotationsList renders correctly 1`] = `
<ul
  className="collection with-header"
  style={
    Object {
      "margin": 0,
    }
  }
>
  <li
    className="collection-header"
  >
     
    <h3>
      Annotations
    </h3>
  </li>
  <div
    style={
      Object {
        "maxHeight": 250,
        "overflow": "auto",
      }
    }
  />
</ul>
`;
